eth-contracts/contracts/EthRewardsManager.sol

```diff
@@ -0,0 +1,166 @@
+pragma solidity ^0.5.0;
+
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/utils/Address.sol";
+import "./InitializableV2.sol";
+import "./Governance.sol";
+
+interface Wormhole {
+    function lockAssets(
+        address asset,
+        uint256 amount,
+        bytes32 recipient,
+        uint8 target_chain,
+        uint32 nonce,
+        bool refund_dust
+    ) external;
+}
+
+
+contract EthRewardsManager is InitializableV2 {
+    using SafeERC20 for ERC20;
+
+    string private constant ERROR_TOKEN_NOT_CONTRACT = (
+        "EthRewardsManager: token is not a contract"
+    );
+    string private constant ERROR_WORMHOLE_NOT_CONTRACT = (
+        "EthRewardsManager: wormhole is not a contract"
+    );
+    string private constant ERROR_ONLY_GOVERNANCE = (
+        "EthRewardsManager: Only governance"
+    );
+
+    address private governanceAddress;
+
+    /// @dev ERC-20 token that will be used to stake with
+    ERC20 internal audiusToken;
+    Wormhole internal wormhole;
+    bytes32 internal recipient;
+
+    address public antiAbuseOracleAddress;
```

Comment:
let's merge this PR first, but @sddioulde can you look into making this a "thrupple"

Reply:
throupple?